homeassistant/components/mystrom/__init__.py

```diff
@@ -1 +1,79 @@
-"""The mystrom component."""
+"""The myStrom integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aiohttp
+from pymystrom.bulb import MyStromBulb
+from pymystrom.exceptions import MyStromConnectionError
+from pymystrom.switch import MyStromSwitch
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def get_device_info(host: str) -> dict[str, Any]:
+    """Get the device info of a myStrom device."""
+    async with aiohttp.ClientSession() as session:
+        async with session.get(f"http://{host}/api/v1/info") as response:
+            if response.status != 200:
+                async with session.get(f"http://{host}/info.json") as response:
+                    if response.status != 200:
+                        raise MyStromConnectionError()
+            content_type = response.headers.get("Content-Type", "")
+            if "application/json" in content_type:
+                return await response.json()
+            raise MyStromConnectionError()
```

Comment:
I'm thinking that this whole `get_device_info` should maybe be a helper method directly in the library.
We shouldn't be making direct URL hits I think.

Reply:
I created a [PR ](https://github.com/home-assistant-ecosystem/python-mystrom/pull/40) in the repository of the library...

Reply:
@fabaff Do you have any recommendation on how to proceed with this?

Reply:
`get_device_info` is now part of the library